Add Percent Change

ag3722 opened the pull request:
- Added calculating insights/metrics for previous transactions by filter
- Added calculating percent difference between current and previous transactions
- Display difference on insight/metric cards

ag3722 committed:
Add Percent Change

- Added calculating insights/metrics for previous transactions by filter
- Added calculating percent difference between current and previous transactions
- Display difference on insight/metric cards

SaifA21 reviewed (commented):
![image](https://github.com/user-attachments/assets/3671e4b6-f3b5-4340-9d3f-220ceff2a51e)
the graphs seem to only display single data points i.e current month 


![image](https://github.com/user-attachments/assets/22821c06-f041-4732-8c12-4d59a7d3b3c2)
maybe we should multiply points redeemed by -1 since it looks weird having an upward arrow for negative

ag3722 committed:
Update Home Page Graph

- Updated labelling and grouping transactions to display on line chart
- Updating filtering transaction by date
- Updated chart styling
- Added legend to chart
